lib/ansible/cli/__init__.py

```diff
@@ -575,6 +575,13 @@ def parse(self):
 
         self.options, self.args = self.parser.parse_args(self.args[1:])
 
+        # --limit must receive non-zero length string
+        if hasattr(self.options, 'subset') and  \
+                self.options.subset != C.DEFAULT_SUBSET and \
+                isinstance(self.options.subset, string_types) and \
+                len(self.options.subset) == 0:
+            raise AnsibleOptionsError("Limit option requires non-zero length string")
+
```

Comment:
I might go for a simpler expression:
```python
limit = getattr(self.options, 'subset', None)
if not limit and limit != C.DEFAULT_SUBSET:
```

Reply:
In ansible.cli.CLI.base_parser, OptParser has : add_option('-l','-limit'...default=C.DEFAULT_SUBSET...
I prefer:
limit = getattr(self.options, 'subset', C.DEFAULT_SUBSET)
if limit != C.DEFAULT_SUBSET and not limt: ( this excludes : "", [], {}, None and -1)